src/NLog/MappedDiagnosticsLogicalContext.cs

```diff
@@ -111,10 +107,7 @@ public static string Get(string item, IFormatProvider formatProvider)
         public static object GetObject(string item)
         {
             object value;
-
-            if (!LogicalThreadDictionary.TryGetValue(item, out value))
-                value = null;
-
+            GetLogicalThreadDictionary().TryGetValue(item, out value);
```

Comment:
This was not my but Resharper refactoring suggestion. 

Actually the value as null is guaranteed when function TryGet returns false:
> "...   otherwise [if the key is not found], the default value for the type of the value parameter [is returned]." - https://msdn.microsoft.com/en-us/library/bb347013(v=vs.110).aspx

As the default value for "object" is null - this refactoring seems valid.